go/worker/common/config.go

```diff
@@ -188,17 +191,34 @@ func NewConfig(ias ias.Endpoint) (*Config, error) {
 		}
 
 		// Configure runtimes.
+		runtimeSignatures := viper.GetStringMapString(CfgRuntimeSignatures)
 		rh.Runtimes = make(map[common.Namespace]runtimeHost.Config)
 		for runtimeID, path := range viper.GetStringMapString(CfgRuntimePaths) {
 			var id common.Namespace
 			if err := id.UnmarshalHex(runtimeID); err != nil {
 				return nil, fmt.Errorf("bad runtime identifier '%s': %w", runtimeID, err)
 			}
 
-			rh.Runtimes[id] = runtimeHost.Config{
+			runtimeHostCfg := runtimeHost.Config{
 				RuntimeID: id,
 				Path:      path,
 			}
+
+			// This config is SGX specific, but that's all that's supported
+			// right now that needs this anyway, the non-SGX provisioner
+			// currently ignores this.
+			if sigPath := runtimeSignatures[runtimeID]; sigPath != "" {
+				runtimeHostCfg.Extra = &hostSgx.RuntimeExtra{
+					SignaturePath: sigPath,
+				}
+			} else {
+				// HACK HACK HACK: Allow dummy SIGSTRUCT generation.
+				runtimeHostCfg.Extra = &hostSgx.RuntimeExtra{
+					UnsafeDebugGenerateSigstruct: true,
```

Comment:
Require don't blame oasis? I know that if don't blame oasis is not set, getting the dummy key will fail anyway, but maybe do it here for good measure?